system/lib/libc/musl/src/thread/__timedwait.c

```diff
@@ -41,8 +41,12 @@ int __timedwait_cp(volatile int *addr, int val,
 
 #ifdef __EMSCRIPTEN__
 	double msecsToSleep = top ? (top->tv_sec * 1000 + top->tv_nsec / 1000000.0) : INFINITY;
-	int is_main_thread = emscripten_is_main_browser_thread();
-	if (is_main_thread || pthread_self()->cancelasync == PTHREAD_CANCEL_ASYNCHRONOUS) {
+	// TODO(kleisauke): Somehow we need to check for emscripten_is_main_runtime_thread() instead of emscripten_is_main_browser_thread().
+	int is_main_runtime_thread = emscripten_is_main_runtime_thread();
+	// cp suffix in the function name means "cancellation point", so this wait can be cancelled
+	// by the users unless current threads cancelability is set to PTHREAD_CANCEL_DISABLE
+	// which may be either done by the user of __timedwait() function.
+	if (is_main_runtime_thread || pthread_self()->canceldisable != PTHREAD_CANCEL_DISABLE) {
```

Comment:
Does doesn't makes sense to me.   Only the main thread should need to busy wait right?   The main runtime thread, if running off the main thread should be able to block.

Reply:
Yes, only the main browser thread should busy wait. Somehow when changing it to `emscripten_is_main_browser_thread()` makes some posix tests stuck on Node (i.e. they never return anything). I still need to investigate why, hence the added TODO item:
https://github.com/kleisauke/emscripten/blob/f4c49eb7bdb36eaebefcca9d5a41c387d415d5ca/system/lib/libc/musl/src/thread/__timedwait.c#L44-L45

Reply:
I've dropped this commit for now, will investigate it later (I'm not sure if `emscripten_is_main_browser_thread()` should return `true` on Node).

Reply:
We need to do busy waiting on side threads in __timedwait_cp function. Because this function may be called from function which is cancellation point (see https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html point "2.9.5 Thread Cancellation").

As we don't have signals we need to check if request to cancel thread hasn't occurred during call to functions like pthread_cond_wait .